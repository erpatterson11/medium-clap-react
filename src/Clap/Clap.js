import React, { Component } from 'react'


export default class Clap extends Component {
    constructor(props) {
        super(props)
        this.state = {
            count: 0
        }
        this.increaseCount = this.increaseCount.bind(this)
    }

    increaseCount() {
        this.setState({count: this.state.count++})
    }

    render() {
        const { count } = this.state
        return (
            <div>  
                <button name="increaseCount" onClick={this.increaseCount}>Count up!!</button>
                <button>Coutn down!!</button>
                <p>{count}</p>
            </div>
        )
    }
}