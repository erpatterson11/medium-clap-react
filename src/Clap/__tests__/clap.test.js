import React from 'react';
import Clap from '../Clap';
import renderer from 'react-test-renderer';

test('Clap increases the count when button is clicked', () => {
  const component = renderer.create(
    <Clap />,
  );
  // renders the component
  let tree = component.toJSON();
    console.log(tree);

  expect(tree).toMatchSnapshot();
  
  let increaseButton = tree.children.filter(val => val.props.name == 'increaseCount')
  // manually invoke the click method
  increaseButton.increaseCount()
  // re-render the component
  tree = component.toJson()
  expect(tree).toMachSnapshot()

});