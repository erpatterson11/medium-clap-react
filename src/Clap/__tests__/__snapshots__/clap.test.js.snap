// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Clap increases the count when button is clicked 1`] = `
<div>
  <button>
    Count up!!
  </button>
  <button>
    Coutn down!!
  </button>
  <p>
    0
  </p>
</div>
`;
